src/Query/Filter/FolderField.ts

```diff
@@ -0,0 +1,36 @@
+import { Group } from '../Group';
+import type { Task } from '../../Task';
+import type { GrouperFunction } from '../Grouper';
+import { TextField } from './TextField';
+
+export class FolderField extends TextField {
+    public fieldName(): string {
+        return 'folder';
+    }
+
+    public value(task: Task): string {
+        const path = task.path;
+        const fileNameWithExtension = task.filename + '.md';
+        const folder = path.substring(0, path.lastIndexOf(fileNameWithExtension));
+        if (folder === '') {
+            return '/';
+        }
+        return folder;
+    }
+
+    public supportsGrouping(): boolean {
+        return true;
+    }
+
+    public grouper(): GrouperFunction {
+        return (task: Task) => {
+            const path = task.path;
+            const fileNameWithExtension = task.filename + '.md';
+            const folder = path.substring(0, path.lastIndexOf(fileNameWithExtension));
+            if (folder === '') {
+                return ['/'];
+            }
+            return [Group.escapeMarkdownCharacters(folder)];
+        };
+    }
+}
```

Comment:
I am uncomfortable with the amount of duplicated code between value() and grouper().